src/signature.rs

```diff
@@ -306,6 +306,27 @@ pub trait VerificationAlgorithm: Sync + private::Private {
 /// Verify the signature `signature` of message `msg` with the public key
 /// `public_key` using the algorithm `alg`.
 ///
+/// To generate the public key from your DER-encoded private key, use
```

Comment:
Since this is RSA-specific, let's put this text in the RSA example below, after the sentence ”RSA signing (but not verification) requires the rsa_signing feature to be enabled” and before the example code block.

I suggest using the wording like this:

> Note that by default OpenSSL writes RSA public keys in SubjectPublicKeyInfo format, not RSAPublicKey format, and by default it Base64-encodes them (”PEM” format).
>
> To convert the PEM SubjectPublicKeyInfo format to the binary RSAPublicKey format needed by `verify()`, use:
> ```sh <example code #2>```
> To extract the RSAPublicKey-formatted public key from an ASN.1 (binary) DER-encoded RSAPrivateKey format private key file, use:
>```sh <example code #1>```

In particular, I think it makes more sense to document how to convert from the default OpenSSL format to the RSAPublicKey format first.

WDYT?